db/httpdb.js

```diff
@@ -439,6 +440,52 @@ module.exports = function (
     )
   }
 
+  DB.prototype.lockAccount = function (account) {
+    var unlockCode = crypto.randomBytes(16).toString('hex');
+    log.trace({ op: 'DB.lockAccount', uid: account && account.uid, unlockCode: unlockCode })
+
+    return this.pool.post(
+      '/account/' + account.uid.toString('hex') + '/lock',
+      {
+        lockedAt: Date.now(),
+        unlockCode: unlockCode
+      }
+    )
+  }
+
+  DB.prototype.unlockAccount = function (account) {
+    log.trace({ op: 'DB.unlockAccount', uid: account && account.uid })
+    return this.pool.post(
+      '/account/' + account.uid.toString('hex') + '/unlock'
+      )
+      .then(null,
+        function (err) {
+          if (err.statusCode === 404) {
+            err = error.unknownAccount()
```

Comment:
There are now two PRs for both the db-server and the db-mem backend to make this not fail if the account doesn't exist. Once they're merged I figure this could be changed and then we can merge this PR. Hope that's ok to all concerned.